.travis.yml

```diff
@@ -0,0 +1,28 @@
+git:
+  depth: false  # fetch all branches
+language: python
+services: [docker]
+matrix:
+  include:
+    - env: TARGET=astyle/Dockerfile.tested
+    - env: TARGET=autopep8/Dockerfile.tested
+    - env: TARGET=black/Dockerfile.tested
+    - env: TARGET=brittany/Dockerfile.tested
+    - env: TARGET=dfmt/Dockerfile.tested
+    - env: TARGET=elm-format/Dockerfile.tested
+    - env: TARGET=google-java-format/Dockerfile.tested
+    - env: TARGET=hindent/Dockerfile.tested
+    - env: TARGET=hlint/Dockerfile.tested
+    - env: TARGET=pg_format/Dockerfile.tested
+    - env: TARGET=php-cs-fixer/Dockerfile.tested
+    - env: TARGET=prettier/Dockerfile.tested
+    - env: TARGET=reorder-python-imports/Dockerfile.tested
+    - env: TARGET=rubocop/Dockerfile.tested
+    - env: TARGET=rustfmt/Dockerfile.tested
+    - env: TARGET=shfmt/Dockerfile.tested
+    - env: TARGET=stylish-haskell/Dockerfile.tested
+    - env: TARGET=terraform/Dockerfile.tested
+    - env: TARGET=yapf/Dockerfile.tested
+    - env: TARGET=generate-travis
```

Comment:
Is this needed? Are you intentionally CI-ing your CI-code-gen?

Reply:
it's so when people add / remove -- this ensures that CI stays up to date

Reply:
You'll have to forgive me for some Travis-noob-ness... To me it looks like each PR will run a job that calls the generate-travis script, but the output of that generation doesn't really do anything, right? The committed version is what is actually used for those jobs and the updated content that this target might make never gets itself committed... right?

Reply:
correct, but it exits nonzero if changes are made which'll fail the PR

Reply:
> it exits nonzero if changes are made

Oh perfect. That makes sense.

If I remember correctly, there's a way in Circle to tell it to execute something to generate the config, rather than actually committing a config, then it will directly run based on what gets generated as of that commit. That would be ideal here, I'm guessing Travis doesn't support it?

Reply:
hmmm I don't recall anything like that in either circle (though admittedly my usage of circle is rather limited) -- I know there isn't something like that in travis though -- purely declarative

Reply:
Ah, it was [Buildkite](https://buildkite.com/docs/pipelines/defining-steps#dynamic-pipelines).

Reply:
ah yep buildkite! we use that for a few things at lyft :D